packages/mui-base/src/Slider/SliderThumb/SliderThumb.tsx

```diff
@@ -0,0 +1,138 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { getStyleHookProps } from '../../utils/getStyleHookProps';
+import { resolveClassName } from '../../utils/resolveClassName';
+import { useRenderPropForkRef } from '../../utils/useRenderPropForkRef';
+import { useSliderContext } from '../Root/SliderProvider';
+import { SliderThumbProps } from './SliderThumb.types';
+import { useSliderThumb } from './useSliderThumb';
+
+function defaultRender(props: React.ComponentPropsWithRef<'span'>) {
+  return <span {...props} />;
+}
+
+function NOOP() {}
+
+const SliderThumb = React.forwardRef(function SliderThumb(
+  props: SliderThumbProps,
+  forwardedRef: React.ForwardedRef<HTMLSpanElement>,
+) {
+  const {
+    render: renderProp,
+    className,
+    children,
+    disabled: disabledProp = false,
+    id,
+    onFocus = NOOP,
+    onBlur = NOOP,
+    onKeyDown = NOOP,
+    ...otherProps
+  } = props;
+
+  const render = renderProp ?? defaultRender;
+
+  const mergedRef = useRenderPropForkRef(render, forwardedRef);
+
+  const { ownerState, active: activeIndex } = useSliderContext('Thumb');
+
+  const { getRootProps, getThumbInputProps, disabled, index } = useSliderThumb({
+    disabled: disabledProp,
+    id,
+    onFocus,
+    onBlur,
+    onKeyDown,
+    rootRef: mergedRef,
+  });
+
+  const styleHooks = React.useMemo(
+    () => getStyleHookProps({ disabled, dragging: activeIndex === index }),
+    [activeIndex, disabled, index],
+  );
+
+  const thumbProps = getRootProps({
+    ...styleHooks,
+    ...otherProps,
+    className: resolveClassName(className, ownerState),
+  });
+
+  const inputProps = getThumbInputProps({ disabled });
+
+  return (
+    <span {...thumbProps}>
+      {children}
+      <input {...inputProps} />
+    </span>
```

Comment:
I still need to come back to this to make it work with the `render` prop somehow 🤔